mixer/adapter/kubernetesenv/cache.go

```diff
@@ -144,11 +145,21 @@ func (c *controllerImpl) Pod(podKey string) (*v1.Pod, bool) {
 		return nil, false
 	}
 	if len(objs) > 0 {
-		pod, ok := objs[0].(*v1.Pod)
-		if !ok {
-			return nil, false
+		pods := []*v1.Pod{}
+		for _, obj := range objs {
+			pod, ok := obj.(*v1.Pod)
+			if !ok {
+				return nil, false
+			}
+			pods = append(pods, pod)
 		}
-		return pod, true
+		// If Pods associated with completed Jobs exist, there can be a case where
+		// more than 1 Pod is found during lookup, and we should always pick the
+		// latest created Pod out of the lot.
```

Comment:
This is ok, but if you are looking for “max(creationTime)” there is no need to sort.
You can do a single pass.

Reply:
Addressed in f500652.